app/stac_api/models.py

```diff
@@ -277,37 +402,29 @@ def __str__(self):
     def clean(self):
         # very simple validation, raises error when geoadmin_variant strings contain special
         # characters or umlaut.
-        if not bool(re.search('^[a-zA-Z0-9]*$', self.geoadmin_variant)):
-            raise ValidationError(_('Property geoadmin:variant not correctly specified.'))
+        try:
+            if not bool(re.search('^[a-zA-Z0-9]*$', self.geoadmin_variant)):
+                raise ValidationError(_("Property geoadmin:variant not compatible with the " +
+                                        "naming conventions."))
+
+        except (KeyError, IndexError) as err:
+            logger.error(_(
+                "Error when trying to access property geoadmin:variant from within asset's " +
+                "clean function: %s"),
+                         err
+                        )
+            raise ValidationError(_(
+                "Error when trying to access property geoamin:variant from within asset's " +
+                "clean function"
+            ))
```

Comment:
Remove the `try ... except` which is not needed here. And add a log in the case of validation error

Reply:
Done